pkg/util/containers/v2/metrics/collector_ecs_fargate.go

```diff
@@ -110,13 +110,13 @@ func convertEcsStats(ecsStats *v2.ContainerStats) *ContainerStats {
 
 	return &ContainerStats{
 		Timestamp: time.Now(),
-		CPU:       convertCPUStats(&ecsStats.CPU),
-		Memory:    convertMemoryStats(&ecsStats.Memory),
-		IO:        convertIOStats(&ecsStats.IO),
+		CPU:       convertEcsCPUStats(&ecsStats.CPU),
+		Memory:    convertEcsMemoryStats(&ecsStats.Memory),
+		IO:        convertEcsIOStats(&ecsStats.IO),
 	}
 }
 
-func convertCPUStats(cpuStats *v2.CPUStats) *ContainerCPUStats {
+func convertEcsCPUStats(cpuStats *v2.CPUStats) *ContainerCPUStats {
```

Comment:
Let's also rename it `convertDockerCPUStats` in the docker collector, wondering if we should have each collector in a separate package.

Reply:
Let me move each collector to a dedicated sub-package. That sounds cleaner.

Reply:
we can do it after https://github.com/DataDog/datadog-agent/pull/9749 is merged

Reply:
+1 to moving each collector to its dedicated sub-package. We can do it in a separate PR if you prefer.

Reply:
Note that collectors are not in sub-packages to avoid registration + circular dependencies issues

Reply:
PR updated with all collectors in their dedicated sub-package.
It required to change the visibility of some symbols and to create an additional package to handle registration without circular dependencies. That extra package isn’t super elegant. :-(